server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanSpanCache.java

```diff
@@ -146,4 +146,19 @@ public void store(String tenantId, List<Span> spans, Function<Span, String> cach
                 .filter(span -> !span.serverSpan())
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public List<Span> getTrace(String tenant, String id) {
+        if (id == null) {
+            throw new NullPointerException("Id should not be null!");
+        }
+
+        QueryFactory<?> queryFactory = Search.getQueryFactory(spansCache);
```

Comment:
Just a heads up - I tried using this in the server and found some dependency problems. You will need to create an integration test, with some suitable source span data, to try this mechanism out running in the server, just to be sure.